.github/workflows/CI-build.yml

```diff
@@ -1,82 +1,84 @@
 name: Build
 
-run-name: "Workflow CI/CD Steps: build, unit and integration testing"
+run-name: "Workflow CI/CD Step: cargo build"
 
 on:
   workflow_call:
-    inputs:
-      CACHING_MODE:
-        required: true
-        type: string
-        
-env:
-  IMAGE_NAME: zencash/sc-ci-base
-  IMAGE_TAG: noble_rust-stable_latest
-  DOCKER_COMPOSE_CMD: "docker compose -f ${GITHUB_WORKSPACE}/ci/docker-compose.yml run --rm cargo-container"
-  DOCKER_BUILD_DIR: /build
-  DOCKER_CARGO_HOME: /tmp/.cargo
-  CMAKE_INSTALL: true
-  LLD_INSTALL: true
+  workflow_dispatch:
 
 jobs:
-  build:
-    runs-on:
-      group: ${{ vars.RUNNER_GROUP }}
+  build-mainchain:
+    runs-on: warp-ubuntu-latest-x64-8x
+    name: Build mainchain
     steps:
-      - name: Git checkout
+      - name: Checkout code
         uses: actions/checkout@v4
 
-      - name: Set up deps cache
-        run: mkdir deps
-
-      - name: Restore main cache
-        if: ${{ !env.ACT && inputs.CACHING_MODE == 'CACHE_RESTORE' }}
-        uses: buildjet/cache/restore@v3
+      - name: Solo chain build
+        uses: ./.github/actions/cargo-cmd
         with:
-          path: |
-            ~/.cargo/bin/
-            ~/.cargo/registry/index/
-            ~/.cargo/registry/cache/
-            ~/.cargo/git/db/
-            target/
-            deps/
-          key: build-cache
+          command: "cargo build -p mainchain --release"
+          use_cache: 'yes'
+          cache_key: 'build-mainchain'
+          cmake_install: 'yes'
+          lld_install: 'yes'
 
-      - name: Set docker env vars
-        run: |
-          echo "USER_ID=$(id -u)" >> $GITHUB_ENV
-          echo "GRP_ID=$(id -g)" >> $GITHUB_ENV
+      - name: Upload zkv-node binary artifact
```

Comment:
The binary is pretty big, are we sure we want to upload it ?

Reply:
Of course, but why uploading them ? Can't we save them locally somewhere ?

Reply:
That is the only way to GET the artifact using a DIFF workflow/job. Artifacts space is free but has some limits like single artifact file can not exceed 2 GB and etc. In our case we use default compression level = 6 and its total size is 21MB